### PR TITLE
[Web] Add properties for page title and icon to WebViewState

### PR DESCRIPTION
**Pull request by sampengilly**
Add properties to the WebViewState class to supply any received title and favicon from the loaded page to the caller.

### CODE REVIEW

**Comment by bentrengrove** [comment]
Thanks for the contribution. I'm just wondering if we have to worry about clearing that state on subsequent page loads. Do these callbacks always get called on page load so they will never be stale?

What about if you have loaded a url and then via a compose state change load a data string, would the title and favicon be set back to null?

**Comment by sampengilly** [comment]
Good point, I'll update to reset the title and favicon on page started.

**Comment by bentrengrove** [comment]
You will also need to update the API file which can be done by running `./gradlew metalavaGenerateSignature`

**Comment by bentrengrove** [comment]
Sorry to be a pain, this new state should also have tests. You could add some test html and test it that way, there is already an example in the project of that I believe. Let me know if you need more info.

**Comment by sampengilly** [comment]
I've added a test for the title state fine. However the favicon state is a real challenge. Since the bitmap isn't necessarily loaded by the time `onPageFinished()` is called (and the idling resource resumes)

File: web/src/androidTest/kotlin/com/google/accompanist/web/WebTest.kt [hunk 1]
```diff
@@ -204,17 +206,48 @@ class WebTest {
...
+        lateinit var state: WebViewState
+
+        rule.setContent {
+            state = rememberWebViewStateWithHTMLData(data = TEST_TITLE_DATA)
+            WebTestContent(webViewState = state, idlingResource = idleResource)
+        }
+
+        // Wait for new title
+        onWebView()
+            .check(webMatches(getTitle(), equalTo("A Test Title")))
```
Comment (bentrengrove) [inline web/src/androidTest/kotlin/com/google/accompanist/web/WebTest.kt:220]: Use your const here

**Review by bentrengrove — COMMENTED** [review]
Nice, thank you! Don't worry about that other test for now, I'll see what I can come up with.

Just need to fix up the one small thing and this will be good to go.

File: web/src/androidTest/kotlin/com/google/accompanist/web/WebTest.kt (hunk 1 above)
Comment (sampengilly) [reply web/src/androidTest/kotlin/com/google/accompanist/web/WebTest.kt]: Done

**Review by bentrengrove — APPROVED** [review]
Awesome. Thanks for the contribution!

**Pull request merged**